README.md

```diff
@@ -188,149 +476,514 @@ Options:
+## Usage with TypeScript
 
-```bash
-npm run test:integration
+`fqm-execution` exports custom-defined TypeScript interfaces used within the code to allow for easy integration into other TypeScript projects. The TypeScript files defining these interfaces can be found [here](https://github.com/projecttacoma/fqm-execution/tree/master/src/types).
+
+The TypeScript interfaces can be imported into another project like so:
+
+```typescript
+import { CalculatorTypes, Enums, CQLTypes, ELMTypes } from 'fqm-execution';
 ```
 
-### Checks
+- `CalculatorTypes`: Contains all interfaces used to specify types during calculation (e.g. return types of calculation functions, type of calculation options object)
 
-When contributing new code, ensure that all tests, lint, and prettier checks pass with the following command:
+```typescript
+import { CalculatorTypes } from 'fqm-execution';
 
-```bash
-npm run check
+const calculationOptions: CalculatorTypes.CalculationOptions = {
+  /* type-safe options */
+};
 ```
 
-### Architecture Overview
+- `Enums`: Contains any `enum`s used across the codebase, specifically with respect to population results reporting (e.g. population types, scoring types, truthy/falsy result reporting)
+
+```typescript
+import { Enums } from 'fqm-execution';
 
-![Overview](doc/FQM.png)
+// Check if a population is the code for initial population
+if (somePopulation === Enums.PopulationType.IPP) {
+  // ...
+}
+
+// Check if some measure scoring code is the code for ratio measure scoring
+if (someScoringCode === Enums.MeasureScoreType.RATIO) {
+  // ...
+}
 
-A visual representation of the calculate sequence of the application can be seen below:
+// Check if some population result has a truthy final result
+if (somePopulationFinalResult === Enums.FinalResult.TRUE) {
+  // ...
+}
 
-![Calculate](doc/calculate_Sequence.png)
+// Check if some population result is not relevant for eCQM calculation (i.e. wasn't considered)
+if (somePopulationRelevanceResult === Enums.Relevance.NA) {
+  // ...
+}
+```
 
-## Contributions
+- `CQLTypes`: Low-level types that interface with the `cql-execution` engine. Unlikely to be needed outside of `fqm-execution` itself, but provided anyway for convenience
+- `ELMTypes`: Low-level types that define basic structure of ELM. Does not define the entire ELM logical specification. Unlikely to be needed outside of `fqm-execution` itself, but provided anyway for convenience
+
+# Recipes
+
+## Displaying Highlighted HTML in a React App
+
+A common use case for the highlighted logic HTML described in the [Measure Logic Highlighting section](#measure-logic-highlighting) is displaying it in a web application of some kind. In the case of React trying to render an HTML string directly will not work
+to due React's security measures to prevent Cross Site Scripting by escaping/sanitizing HTML strings that attempt to be rendered in the DOM.
```

Comment:
```suggestion
due to React's security measures to prevent Cross Site Scripting by escaping/sanitizing HTML strings that attempt to be rendered in the DOM.
```

Reply:
This one is embarassing